lib/ansible/module_utils/crypto.py

```diff
@@ -46,3 +48,68 @@ def get_fingerprint(path):
         pass
 
     return fingerprint
+
+
+def load_privatekey(path, passphrase=None):
+    """Load the specified OpenSSL private key."""
+
+    try:
+        privatekey_content = open(path, 'r').read()
+        privatekey = crypto.load_privatekey(crypto.FILETYPE_PEM,
+                                            privatekey_content,
+                                            passphrase)
+        return privatekey
+    except (IOError, OSError) as exc:
+        raise OpenSSLModuleError(exc)
+
+
+def load_certificate(path):
+    """Load the specified certificate."""
+
+    try:
+        cert_content = open(path, 'r').read()
+        cert = crypto.load_certificate(crypto.FILETYPE_PEM, cert_content)
+        return cert
+    except (IOError, OSError) as exc:
+        raise OpenSSLModuleError(exc)
+
+
+class OpenSSLModuleError(Exception):
+    pass
+
+
+class OpenSSLModule(object):
+
+    def __init__(self, module):
+        self.path = module.params['path']
+        self.state = module.params['state']
+        self.force = module.params['force']
+        self.name = os.path.basename(module.params['path'])
+        self.changed = True
+        self.check_mode = module.check_mode
```

Comment:
The usual idiom I see with these *Module type of classes is that they hold onto module in an attribute.  I happen to think that's the wrong thing to do (as it leads to calling self.module.fail_json() instead of raising exceptions that are then handled at a higher level).  But if you aren't going to hold onto module, you shouldn't pass it into the constructor.  Instead pass in the specific parameters that you care about:
``` python
def __init__(self, path, state, force, check_mode):
```